compiler/fir/semantics/src/org/jetbrains/kotlin/fir/resolve/dfa/cfg/CFGNode.kt

```diff
@@ -654,7 +665,8 @@ class ConstExpressionNode(owner: ControlFlowGraph, override val fir: FirConstExp
 
 // ----------------------------------- Check not null call -----------------------------------
 
-class CheckNotNullCallNode(owner: ControlFlowGraph, override val fir: FirCheckNotNullCall, level: Int, id: Int) : CFGNode<FirCheckNotNullCall>(owner, level, id) {
+class CheckNotNullCallNode(owner: ControlFlowGraph, override val fir: FirCheckNotNullCall, level: Int, id: Int)
+    : CFGNode<FirCheckNotNullCall>(owner, level, id), UnionNodeMarker {
```

Comment:
Because it's implemented as a synthetic call so in theory it could be the first completed call and thus be a data flow sink for postponed lambdas inside `x`. I'm not sure if that's possible in practice.

Reply:
Got it. Actually it might me something like that:
```kotlin
val x: Int = select(run { materialize() }, run { materialize() })!!
```